Lib/reprlib.py

```diff
@@ -60,12 +60,26 @@ def repr(self, x):
         return self.repr1(x, self.maxlevel)
 
     def repr1(self, x, level):
-        typename = type(x).__name__
-        if ' ' in typename:
-            parts = typename.split()
-            typename = '_'.join(parts)
-        if hasattr(self, 'repr_' + typename):
-            return getattr(self, 'repr_' + typename)(x, level)
+
+        _lookup = {
+            ("builtins", "tuple"): self.repr_tuple,
+            ("builtins", "list"): self.repr_list,
+            ("array", "array"): self.repr_array,
+            ("builtins", "set"): self.repr_set,
+            ("builtins", "frozenset"): self.repr_frozenset,
+            ("collections", "deque"): self.repr_deque,
+            ("builtins", "dict"): self.repr_dict,
+            ("builtins", "str"): self.repr_str,
+            ("builtins", "int"): self.repr_int,
+        }
+
+        _type = type(x)
+        module = _type.__module__
```

Comment:
Some very weird types might not have `__module__` or it might not be a `str`. Let's ignore them here. Otherwise we would break existing usages of them with `reprlib`.